controllers/datadogagent/feature/enabledefault/feature.go

```diff
@@ -19,13 +30,112 @@ func init() {
 }
 
 func buildDefaultFeature(options *feature.Options) feature.Feature {
-	return &defaultFeature{}
+	return &defaultFeature{
+		credentialsInfo: credentialsInfo{
+			secretCreation: secretInfo{
+				data: make(map[string]string),
+			},
+		},
+		dcaTokenInfo: dcaTokenInfo{
+			secretCreation: secretInfo{
+				data: make(map[string]string),
+			},
+		},
+	}
+}
+
+type defaultFeature struct {
+	namespace string
+	owner     metav1.Object
+
+	credentialsInfo    credentialsInfo
+	dcaTokenInfo       dcaTokenInfo
+	clusterAgent       clusterAgentConfig
+	agent              agentConfig
+	clusterCheckRunner clusterCheckRunnerConfig
+}
+
+type credentialsInfo struct {
+	apiKey         keyInfo
+	appKey         keyInfo
+	secretCreation secretInfo
+}
+
+type dcaTokenInfo struct {
+	token          keyInfo
+	secretCreation secretInfo
+}
+
+type keyInfo struct {
+	SecretName string
+	SecretKey  string
+}
+
+type secretInfo struct {
+	createSecret bool
+	name         string
+	data         map[string]string
+}
+
+type clusterAgentConfig struct {
+	serviceAccountName string
+}
+
+type agentConfig struct {
+	serviceAccountName string
 }
 
-type defaultFeature struct{}
+type clusterCheckRunnerConfig struct {
+	serviceAccountName string
+}
 
 func (f *defaultFeature) Configure(dda *v2alpha1.DatadogAgent) feature.RequiredComponents {
 	trueValue := true
+	f.namespace = dda.Namespace
+	f.owner = dda
+
+	f.clusterAgent.serviceAccountName = v2alpha1.GetClusterAgentServiceAccount(dda)
+	f.agent.serviceAccountName = v2alpha1.GetAgentServiceAccount(dda)
+	f.clusterCheckRunner.serviceAccountName = v2alpha1.GetClusterChecksRunnerServiceAccount(dda)
+
```

Comment:
should we check here if the component is disabled? `dda.Spec.Override[ClusterAgentComponentName].Disabled` or should we still configure this feature and make a decision in the `Override` function?